packages/rule-components/src/RuleTable/RuleTable.js

```diff
@@ -0,0 +1,227 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Table, TableBody, TableHeader, sortable } from '@patternfly/react-table';
+import { Skeleton } from '@redhat-cloud-services/frontend-components/components/Skeleton';
+import { TableToolbar } from '@redhat-cloud-services/frontend-components/components/TableToolbar';
+import { Pagination, PaginationVariant } from '@patternfly/react-core/dist/js/components/Pagination';
+import { PrimaryToolbar } from '@redhat-cloud-services/frontend-components/components/PrimaryToolbar';
+import flatten from 'lodash/flatten';
+import debounce from 'lodash/debounce';
+import { calculateMeta, calculateActiveFilters, createRows } from './helpers';
+
+class RuleTable extends Component {
+    state = {
+        filterValues: {},
+        expanded: []
+    }
+
+    debounceFetchData = debounce(options => {
+        const { fetchData } = this.props;
+        fetchData(options);
+    }, 500)
+
+    onUpdate = (isDebounce = false) => {
+        const { fetchData, rules, columns, sortBy, detail } = this.props;
+        const { filterValues } = this.state;
+        const method = isDebounce ? this.debounceFetchData : fetchData;
+        method({
+            meta: calculateMeta(rules.meta),
+            filterValues,
+            sortBy: { ...sortBy, column: columns[sortBy.index - Boolean(detail)] }
+        });
+    }
+
+    onFilterDelete = (_e, [ deleteItem ], isAll) => {
+        const { filterValues } = this.state;
+        if (isAll) {
+            this.setState({ filterValues: {} }, () => this.onUpdate());
+        } else {
+            if (Array.isArray(filterValues[deleteItem.type])) {
+                const activefilter = filterValues[deleteItem.type];
+                const [ chipToDelete ] = deleteItem.chips;
+                this.setState({ filterValues: {
+                    ...filterValues,
+                    [deleteItem.type]: activefilter.filter((item) => item !== chipToDelete.value)
+                } }, () => this.onUpdate());
+            } else {
+                delete filterValues[deleteItem.type];
+                this.setState({ filterValues }, () => this.onUpdate());
+            }
+        }
+    }
+
+    onCollapse = (_e, key, isOpen, { ruleId }) => {
+        const { expanded } = this.state;
+        if (isOpen) {
+            this.setState({
+                expanded: [
+                    ...expanded,
+                    ruleId || key
```

Comment:
I think better to avoid this construction and define variable that will have needed value, for example:
`const expandedId = ruleId || key`

Reply:
defining extra variable will result in more memory usage (i know its minimal size). If you don't use a value like this multiple times, it's more efficient to do it inline so it will end up on the stack.

Reply:
But it's value is used more then one time.

Reply:
Yup